File: ArrayManip/sol.py
def arrayManipulation(n, queries):
    base = [0] * (n + 2)
    for x in queries:
        base[x[0]] += x[2]
        base[x[1] + 1] -= x[2]
    mx, acc = 0, 0
    for i, val in enumerate(base):
        acc += val
        mx = max(acc, mx)
    return mx

File: ArrayManip/sol.go
package ArrayManip

import "math"

func ArrayManipulation(n int32, queries [][]int32) int64 {
	base := make([]int32, n+2)
	for _, v := range queries {
		base[v[0]] += v[2]
		base[v[1]+1] -= v[2]
	}
	mx, acc := 0, 0
	for _, val := range base {
		acc += int(val)
		mx = int(math.Max(float64(mx), float64(acc)))
	}
	return int64(mx)
}
